infra/Pos/Discovery/Class.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE RankNTypes #-}
+
+-- | Abstraction over possibility to retrieve neighbors.
+
+module Pos.Discovery.Class
+       ( MonadDiscovery (..)
+       ) where
+
+import           Control.Monad.Trans              (MonadTrans)
+import           Pos.Communication.Types.Protocol (NodeId)
+import           Universum
+
+class (Monad m) => MonadDiscovery m where
+    getPeers  :: m (Set NodeId)
+    findPeers :: m (Set NodeId)
```

Comment:
Can you add documentation of these functions? It's not obvious what `findPeers` should return, for example.